contracts/libs/Types.sol

```diff
@@ -21,6 +21,7 @@ library Types {
         uint256 stakeCommitted;
         uint256 finalisesOn;
         uint256 timestamp;
+        BatchType batchType;
```

Comment:
We add BatchType for batches.

Reply:
Yeah that's a good point

Reply:
Is the idea here to have a batchType and its `slashable` if there is a tx in that batch that doesnt belong there?

Reply:
Reposting the conclusion from the offline discussion here. We don't need to check if the tx type matches the batch type explicitly. The sender signs the uncompressed tx with the tx type in it, so when we validate the signature of a compressed tx in the batch, the signature is only valid if the tx type is correct.